openapi_metadata.go

```diff
@@ -0,0 +1,449 @@
+package fuego
+
+import (
+	"encoding/xml"
+	"fmt"
+	"log/slog"
+	"reflect"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+const (
+	metadataParserExample     = "exampleParser"
+	metadataParserValidation  = "validationParser"
+	metadataParserDescription = "descriptionParser"
+	metadataParserXML         = "XMLParser"
+	metadataParserJSON        = "JSONParser"
+)
+
+var DefaultParsers = []MetadataParserEntry{
+	{Name: metadataParserExample, Parser: MetadataParserExample},
+	{Name: metadataParserValidation, Parser: MetadataParserValidation},
+	{Name: metadataParserDescription, Parser: MetadataParserDescription},
+	{Name: metadataParserXML, Parser: MetadataParserXML},
+	{Name: metadataParserJSON, Parser: MetadataParserJSON},
+}
+
+type MetadataParserParams struct {
+	// Reflective information of a struct field
+	Field reflect.StructField
+	// Name of the field
+	FieldName string
+	// OpenAPI schema of the field
+	Property *openapi3.Schema
+	// Reference to the OpenAPI schema
+	SchemaRef *openapi3.SchemaRef
+	// Additional metadata as key-value pairs
+	Additional map[string]interface{}
+}
+
+type MetadataParserFunction func(params MetadataParserParams)
+
+type MetadataParserEntry struct {
+	Name   string
+	Parser MetadataParserFunction
+}
+
+type MetadataParsers struct {
+	registeredParsers []MetadataParserEntry
+	parserLock        sync.Mutex
+	registeredNames   map[string]bool
```

Comment:
From what I've read, `registeredNames` is a Set that just checks that parsers aren't registered twice. Since you just iterate on `registeredParsers`, can you please simply use a map? `registeredParsers map[string]MetadataParserEntry`.